File: ros_basics/scripts/float_publisher.py
#!/usr/bin/env python

import rospy 
from std_msgs.msg import Float32

def publisher():
	rospy.init_node("float32_publisher", anonymous=True)
	pub = rospy.Publisher('float', Float32, queue_size=100)

	rate= rospy.Rate(10)
	var = 0.000

	while not rospy.is_shutdown():
		rospy.loginfo("Publisher sent : %f", var)
		pub.publish(var)
		rate.sleep()

		var = var + 0.05

if __name__ == '__main__':
	try:
		publisher()
	except rospy.ROSInterruptException():
		pass

File: ros_basics/src/float_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Float32.h"
#include <iostream>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "float_publisher");
	ros::NodeHandle nh;
	ros::Publisher pub = nh.advertise<std_msgs::Float32>("/float",10);

	ros::Rate loop_rate(10);

	float var = 0.1;

	while(ros::ok())
	{
		std_msgs::Float32 msg;
		msg.data = var;
		ROS_INFO("Publisher sent: %f", msg.data);
		pub.publish(msg);

		ros::spinOnce();
		loop_rate.sleep();
		var = var + 0.2;
	}
	return 0;
}